### PR TITLE
fix(useDebouncedField): fixing helpers logic

### PR DESCRIPTION
**Pull request by amabarr**
### Overview

1. Fixes `useGetInitialFormValue` to better keep the behavior of useField by setting string fields without a initialValue to `""` by default instead of `undefined`
2. Fixes `useMakeSetFormDirty` logic so that the form can be re-dirtied after a submission. 
3. Updates unit tests

### PR Checklist

- [x] Related to designs: N/A
- [x] Related to JIRA ticket: [CONPLAT-2196](https://codecademy.atlassian.net/browse/CONPLAT-2196)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change (unit tests should cover this)
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

[Don't make me tap the sign.](https://i.imgur.com/sy93D9I.png)

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/32299)  | [Monolith Env](https://pr-32299-monolith.dev-eks.codecademy.com/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/portal-app/pull/2500)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-2500 'Named link title')   |
| Another Repo | [Author](https://github.com/codecademy-engineering/author/pull/2224 'Named link title') | [Author PR](https://author-1.pr.codecademy.com/)  |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.191-alpha.ee15ce.0<br/>@codecademy/gamut-labs@33.1.12-alpha.ee15ce.0<br/>@codecademy/gamut@50.5.4-alpha.ee15ce.0<br/>@codecademy/monaco@0.6.81-alpha.ee15ce.0<br/>@codecademy/styleguide@61.3.5-alpha.ee15ce.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://630cd4a5b736cd52fa509d85--gamut-preview.netlify.app>https://630cd4a5b736cd52fa509d85--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/630cd4a5b736cd52fa509d85>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.191-alpha.c81008.0<br/>@codecademy/gamut-labs@33.1.12-alpha.c81008.0<br/>@codecademy/gamut@50.5.4-alpha.c81008.0<br/>@codecademy/monaco@0.6.81-alpha.c81008.0<br/>@codecademy/styleguide@61.3.5-alpha.c81008.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://630ce72e27e64d06234330f5--gamut-preview.netlify.app>https://630ce72e27e64d06234330f5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/630ce72e27e64d06234330f5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.193-alpha.1ba002.0<br/>@codecademy/gamut-labs@33.2.1-alpha.1ba002.0<br/>@codecademy/gamut@50.5.5-alpha.1ba002.0<br/>@codecademy/monaco@0.6.82-alpha.1ba002.0<br/>@codecademy/styleguide@61.4.1-alpha.1ba002.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://630cea12dea2bd0783655c00--gamut-preview.netlify.app>https://630cea12dea2bd0783655c00--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/630cea12dea2bd0783655c00>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.193-alpha.fd9c40.0<br/>@codecademy/gamut-labs@33.2.1-alpha.fd9c40.0<br/>@codecademy/gamut@50.5.5-alpha.fd9c40.0<br/>@codecademy/monaco@0.6.82-alpha.fd9c40.0<br/>@codecademy/styleguide@61.4.1-alpha.fd9c40.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://630cef02854dd50d10b601f6--gamut-preview.netlify.app>https://630cef02854dd50d10b601f6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/630cef02854dd50d10b601f6>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.193-alpha.b27f24.0<br/>@codecademy/gamut-labs@33.2.1-alpha.b27f24.0<br/>@codecademy/gamut@50.5.5-alpha.b27f24.0<br/>@codecademy/monaco@0.6.82-alpha.b27f24.0<br/>@codecademy/styleguide@61.4.1-alpha.b27f24.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://630cff7f3e4a70048222228f--gamut-preview.netlify.app>https://630cff7f3e4a70048222228f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/630cff7f3e4a70048222228f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.193-alpha.f15019.0<br/>@codecademy/gamut-labs@33.2.1-alpha.f15019.0<br/>@codecademy/gamut@50.5.5-alpha.f15019.0<br/>@codecademy/monaco@0.6.82-alpha.f15019.0<br/>@codecademy/styleguide@61.4.1-alpha.f15019.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://630d0a19cc656e0e3c0811db--gamut-preview.netlify.app>https://630d0a19cc656e0e3c0811db--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/630d0a19cc656e0e3c0811db>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.193-alpha.7b0a19.0<br/>@codecademy/gamut-labs@33.2.1-alpha.7b0a19.0<br/>@codecademy/gamut@50.5.5-alpha.7b0a19.0<br/>@codecademy/monaco@0.6.82-alpha.7b0a19.0<br/>@codecademy/styleguide@61.4.1-alpha.7b0a19.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://630d0e843e4a7011ed2222d2--gamut-preview.netlify.app>https://630d0e843e4a7011ed2222d2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/630d0e843e4a7011ed2222d2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.193-alpha.7e8e0e.0<br/>@codecademy/gamut-labs@33.2.1-alpha.7e8e0e.0<br/>@codecademy/gamut@50.5.5-alpha.7e8e0e.0<br/>@codecademy/monaco@0.6.82-alpha.7e8e0e.0<br/>@codecademy/styleguide@61.4.1-alpha.7e8e0e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://630d11ccb1b0970fa77b693f--gamut-preview.netlify.app>https://630d11ccb1b0970fa77b693f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/630d11ccb1b0970fa77b693f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.193-alpha.b7bb58.0<br/>@codecademy/gamut-labs@33.2.1-alpha.b7bb58.0<br/>@codecademy/gamut@50.5.5-alpha.b7bb58.0<br/>@codecademy/monaco@0.6.82-alpha.b7bb58.0<br/>@codecademy/styleguide@61.4.1-alpha.b7bb58.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://630d2c67cbac8f302122f424--gamut-preview.netlify.app>https://630d2c67cbac8f302122f424--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/630d2c67cbac8f302122f424>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.194-alpha.12d47c.0<br/>@codecademy/gamut-labs@33.2.2-alpha.12d47c.0<br/>@codecademy/gamut@50.6.1-alpha.12d47c.0<br/>@codecademy/monaco@0.6.83-alpha.12d47c.0<br/>@codecademy/styleguide@61.4.2-alpha.12d47c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://630e12ee9d4b8e61dba8a91a--gamut-preview.netlify.app>https://630e12ee9d4b8e61dba8a91a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/630e12ee9d4b8e61dba8a91a>Deploy Logs</a></p>

**Review by ZebGirouard — APPROVED** [review]
I'm curious if non-Author folks utilize `useDebouncedField` (can't find it in `Codecademy` or `portal-app`), but for Author this seems fairly bullet-proof (tried a pretty heavy combination of typing fast, selecting within a field, selecting a range, saving/not-saving, and everything works as-expected).

File: packages/gamut/src/ConnectedForm/__tests__/useDebouncedField.test.tsx [hunk 1]
```diff
@@ -133,15 +135,16 @@ describe('ConnectedForm - useDebouncedField', () => {
       fireEvent.change(input, { target: { value: 'tif' } });
```
[the remark this answers is not in the document]
Comment (dreamwasp) [reply packages/gamut/src/ConnectedForm/__tests__/useDebouncedField.test.tsx:135]: 🤔  yeah i'm not sure what the most up-to-date best practices are atm but i think moving towards `userEvent` makes sense to me

**Review by dreamwasp — APPROVED** [review]
makes sense to me

**Pull request merged**